pkg/kf/commands/apps/scale.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package apps
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/google/kf/pkg/apis/kf/v1alpha1"
+	"github.com/google/kf/pkg/kf/commands/config"
+	"github.com/google/kf/pkg/kf/commands/utils"
+	"github.com/google/kf/pkg/kf/kfapps"
+	"github.com/spf13/cobra"
+)
+
+// NewScaleCommand creates a command capable of scaling an app.
+func NewScaleCommand(
+	p *config.KfParams,
+	client kfapps.Client,
+) *cobra.Command {
+	var (
+		instances    int
+		autoscaleMin int
+		autoscaleMax int
+	)
+
+	var scale = &cobra.Command{
+		Use:   "scale APP_NAME",
+		Short: "Change the instance count for an app.",
+		Example: `
+  kf scale myapp --i 3 # Scale to exactly 3 instances
+  kf scale myapp --instances 3 # Scale to exactly 3 instances
+  kf scale myapp --autoscale-min 3 # Autoscaler won't scale below 3 instances
+  kf scale myapp --autoscale-max 5 # Autoscaler won't scale above 5 instances
+  kf scale myapp --autoscale-min 3 --autoscale-max 5 # Autoscaler won't below 3 or above 5 instances
+  `,
+		Args: cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := utils.ValidateNamespace(p); err != nil {
+				return err
+			}
+
+			if instances < 0 && autoscaleMin < 0 && autoscaleMax < 0 {
+				return errors.New("--instances, --autoscaleMin, or --autoscaleMax flag are required")
+			}
+
+			if instances >= 0 && (autoscaleMin >= 0 || autoscaleMax >= 0) {
+				return errors.New("exact scaling (--instances) and autoscale can not be set.")
+			}
+
+			if autoscaleMax < autoscaleMin {
+				return errors.New("--autoscale-min must be less than --autoscale-max")
+			}
+
+			appName := args[0]
+
+			cmd.SilenceUsage = true
+
+			mutator := func(app *v1alpha1.App) error {
+				app.Spec.Instances.Min = nil
+				app.Spec.Instances.Max = nil
+				app.Spec.Instances.Exactly = nil
+
```

Comment:
WDYT about returning `app.Spec.Instances.Validate(context.Background())` rather than repeating the validation logic above? Right now I think it's in three places which is less than ideal...

Reply:
So honestly I would rather let the webhook do the validation. The problem I'm having is that I can't get Watch to work with App. I think for the moment, I'll just leave the manual validation and rip it out once we get the watch stuff sorted out.

Reply:
Meh, I thought about it more and changed my mind.